manager/manager_test.go

```diff
@@ -2,19 +2,283 @@ package manager
+	meta := metadata.Metadata{
+		VolumeID:   "vol-id",
+		TargetPath: "/fake/path",
+	}
+	store.RegisterMetadata(meta)
+	// Ensure we stop managing the volume after the test
+	defer func() {
+		store.RemoveVolume(meta.VolumeID)
+		m.UnmanageVolume(meta.VolumeID)
+	}()
+
+	// Attempt to issue the certificate & put it under management
+	managed, err := m.ManageVolumeImmediate(ctx, meta.VolumeID)
+	if !managed {
+		t.Errorf("expected management to have started, but it did not")
+	}
+	if err != nil {
+		t.Errorf("expected no error from ManageVolumeImmediate but got: %v", err)
+	}
+
+	// Assert the certificate is under management
+	if !m.IsVolumeReady(meta.VolumeID) {
+		t.Errorf("expected volume to be marked as Ready but it is not")
+	}
+}
+
+func TestManager_PropagatesRequestConditionMessages(t *testing.T) {
+	tests := []struct {
+		approved      string
+		ready         string
+		expectedError string
+	}{
+		{
+			approved:      "",
+			ready:         "",
+			expectedError: "waiting for request: request \"certificaterequest-name\" has not yet been approved by approval plugin",
+		},
+		{
+			approved:      "",
+			ready:         "pending",
+			expectedError: "waiting for request: request \"certificaterequest-name\" has not yet been approved by approval plugin",
+		},
+		{
+			approved:      "",
+			ready:         "failed",
+			expectedError: "waiting for request: request \"certificaterequest-name\" has failed: failed",
+		},
+		//"pending approval, ready == true": {}
+		{
+			approved:      "denied",
+			ready:         "",
+			expectedError: "waiting for request: request \"certificaterequest-name\" has been denied by the approval plugin: denied",
+		},
+		{
+			approved:      "denied",
+			ready:         "pending",
+			expectedError: "waiting for request: request \"certificaterequest-name\" has been denied by the approval plugin: denied",
+		},
+		{
+			approved:      "denied",
+			ready:         "failed",
+			expectedError: "waiting for request: request \"certificaterequest-name\" has been denied by the approval plugin: denied",
+		},
+		//"denied, ready == true": {},
+		{
+			approved:      "approved",
+			ready:         "",
+			expectedError: "waiting for request: request \"certificaterequest-name\" has no ready condition",
+		},
+		{
+			approved:      "approved",
+			ready:         "pending",
+			expectedError: "waiting for request: request \"certificaterequest-name\" is pending: pending",
+		},
+		{
+			approved:      "approved",
+			ready:         "failed",
+			expectedError: "waiting for request: request \"certificaterequest-name\" has failed: failed",
+		},
+		//"approved, ready == true": {},
+	}
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("approval=%q, readiness=%q", test.approved, test.ready), func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+			defer cancel()
+
+			opts := newDefaultTestOptions(t)
+			m, err := NewManager(opts)
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
I'm personally not much of a fan of using the assert and require library, but that's just me 🤷‍♂️ I find leaning to stdlib makes things really obvious for anyone that isn't well versed with these libraries... I see we currently only using testify in one place: https://github.com/cert-manager/csi-lib/blob/fe9b5eff41b7a867e93102b4e7cf7847b40aa23c/manager/util/tokenrequest_test.go#L26 - given it does also increase the size of consumer's dependency graphs I'd lean towards removing it altogether rather than using it more often

Reply:
Make sense, I'm fine with that :+1: